lib/mercury/received_message.rb

```diff
@@ -35,6 +36,11 @@ def nack
       metadata.reject(requeue: true)
     end
 
+    def republish(&k)
+      k ||= proc{}
```

Comment:
Ah, nice trick for optional block when delegating that block to another method....